File: backend/app/models/discount_code.py
from datetime import datetime

from sqlalchemy import Boolean, Column, Integer, String, DateTime

from app.db.base_class import Base



class DiscountCode(Base):
    __tablename__ = 'discount_code'
    
    id = Column(Integer, primary_key=True, index=True)
    code = Column(String, nullable=False)
    discount = Column(Integer, nullable=False)
    state = Column(Boolean, nullable=False)
    expiration_date = Column(DateTime, nullable=True)
    
    created_at = Column(DateTime, nullable=True, default=datetime.utcnow())
    updated_at = Column(DateTime, nullable=True, default=datetime.utcnow())

    
    

File: itsocks-laravel/app/Models/DiscountCode.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class DiscountCode extends Model
{
    protected $table = 'discount_code';

    protected $fillable = [
        'code',
        'discount_type',
        'discount',
        'state',
        'expiration_date',
    ];

    protected $casts = [
        'discount' => 'integer',
        'state' => 'boolean',
        'expiration_date' => 'datetime',
    ];

    public function scopeActive($query)
    {
        return $query->where('state', true)
            ->where(function ($q) {
                $q->whereNull('expiration_date')
                    ->orWhere('expiration_date', '>', now());
            });
    }
}
